### PR TITLE
Refactor Spring completion to not implement it's own Rake completion

### PR DESCRIPTION
**Pull request by MasterRoot24**
The Spring completion script implemented it's own Rake completion logic. I've changed it to use the Rake completion logic provided by homesick-completions.

### CODE REVIEW

**Comment by bfontaine** [comment]
Do you mind making a pull-request to upstream instead?

**Comment by MasterRoot24** [comment]
@bfontaine I can do, although I have changed the logic of the script considerably and I'm not sure that's what upstream would want. The aim with my fork is that it depends on other `bash-completion` functions instead of implementing it's own.

**Comment by bfontaine** [comment]
@MasterRoot24 Thanks, in that case please add a dependency on the `rake-completion` formula on that one.

**Comment by MasterRoot24** [comment]
Of course, that would be a good idea wouldn't it? Why didn't I think of that before opening the PR? haha 😜 

I've just pushed a fix with that dependency in there now.

File: spring-completion.rb [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
-  homepage "https://github.com/jacaetevha/spring_bash_completion"
-  url "https://github.com/jacaetevha/spring_bash_completion/archive/v0.0.1.tar.gz"
-  sha256 "a97b256dbdaca894dfa22bd96a6705ebf4f94fa8206d05f41927f062c3dd60bf"
-  head "https://github.com/jacaetevha/spring_bash_completion.git"
-
+  homepage "https://github.com/MasterRoot24/spring_bash_completion"
+  url "https://github.com/MasterRoot24/spring_bash_completion/archive/v0.0.2.tar.gz"
+  sha256 "3351473ffb0a76aaabf6967757a2d84541ad34e42cfb91cca5ca57552880c92a"
   bottle :unneeded
+  depends_on "rake-completion"
```
Comment (bfontaine) [inline spring-completion.rb]: Please leave one empty line before and after the `bottle :unneeded` line.

**Comment by MasterRoot24** [comment]
@bfontaine I've just pushed a fix containing the new lines you requested.

I've FYI also added a `head` back in, pointing to the GitHub repo.

**Comment by MasterRoot24** [comment]
Hi @bfontaine Is there anything that you're waiting for from me on this? I'd like to move it forward if possible, please 🙂

**Comment by MasterRoot24** [comment]
On reflection, I've changed my mind with this.

Due to my changes essentially replacing Spring completion and changing how it works quite a bit, I've decided to not pursue getting this merged into Homebrew/homebrew-completions and I've spun off my own tap where I will host my version of the Formula for anyone to use if they want.

My tap can be found [here](https://github.com/MasterRoot24/homebrew-shell-completions).

I feel this is better for the community, who can carry on using the Homebrew provided Formula and shell completion script in this repo, but can also opt for my alternative if they prefer.

**Comment by MasterRoot24** [comment]
I'll leave my branch in my fork, so if the community wants this to be merged in the future, we can do.

**Comment by bfontaine** [comment]
I think this is a good decision. Sorry for the delay on this, @MasterRoot24!

**Pull request closed without merging**